project/BRANCHES-AND-TAGS.md

```diff
@@ -0,0 +1,35 @@
+Branches and tags
+=================
+
+Note: details of the release process for the Engine are documented in the
+[RELEASE-CHECKLIST](https://github.com/docker/docker/blob/master/project/RELEASE-CHECKLIST.md).
+
+# Branches
+
+Docker projects should normally have only three living branches at all time, including the regular
+`master` branch:
+
+## `docs` branch
+
+The `docs` branch support documentation updates between product releases. This branch allow us to
+decouple documentation releases from product releases.
+
+## `release` branch
```

Comment:
Typically, we have a release branch per major version. For instance, we have `release/2.0` and we'll have `release/2.1`. This allows doing patch releases on individual branches.